tests/unit_tests/linear_solver/NNLS_tests.cpp

```diff
@@ -667,9 +663,8 @@ bool case_1 (const PHiLiP::Parameters::AllParameters *const all_parameters,
   A_eig << 1, 1,  2, 4,  3, 9,  4, 16;
   b_eig << 0.6, 2.2, 4.8, 8.4;
   x_eig << 0.1, 0.5;
-  double b_pt[] = {0.6, 2.2, 4.8, 8.4};
```

Comment:
I'm not familiar with this part of the code, but can you explain why you're able to remove `b_pt`?

Reply:
Of course, b_pt was only used to construct Epetra_Vector b using the following line
`Epetra_Vector b(Copy, A.RowMap(), b_pt);`
However, if A.RowMap() streches over more than one core, it will only copy the number of local elements from b_pt starting from 0. The resulting vector will have the following data and format. 
<code>Epetra::Vector\
Vector b\
MyPID&nbsp;GID&nbsp;Value\
0&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;0&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;0.6\
0&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;1&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;2.2\
1&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;2&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;0.6\
1&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;3&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;2.2
</code>
To correct this, the helper function eig_to_epetra_vector corrects this issue to give the following Epetra_Vector. b_pt is no longer needed and was deleted from NNLS_tests.cpp.
<code>Epetra::Vector\
Vector b\
MyPID&nbsp;GID&nbsp;Value\
0&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;0&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;0.6\
0&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;1&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;2.2\
1&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;2&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;4.8\
1&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;3&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;8.4
</code>